stablehlo/conversions/tosa/transforms/StablehloQuantLegalizeToTosaRescale.cpp

```diff
@@ -170,20 +304,115 @@ LogicalResult matchAndRewriteAddSub(StablehloOp op, PatternRewriter& rewriter) {
   return success();
 }
 
+LogicalResult matchAndRewriteOp(stablehlo::AddOp op,
+                                PatternRewriter &rewriter) {
+  return matchAndRewriteBinaryOp(op, rewriter, GetAddSubRescaleScales);
+}
+
+LogicalResult matchAndRewriteOp(stablehlo::SubtractOp op,
+                                PatternRewriter &rewriter) {
+  return matchAndRewriteBinaryOp(op, rewriter, GetAddSubRescaleScales);
+}
+
+LogicalResult matchAndRewriteOp(stablehlo::MulOp op,
+                                PatternRewriter &rewriter) {
+  return matchAndRewriteBinaryOp(op, rewriter, GetMulDivRescaleScales);
+}
+
+LogicalResult matchAndRewriteOp(stablehlo::DivOp op,
+                                PatternRewriter &rewriter) {
+  return matchAndRewriteBinaryOp(op, rewriter, GetMulDivRescaleScales);
+}
+
+LogicalResult matchAndRewriteOp(stablehlo::MinOp op,
+                                PatternRewriter &rewriter) {
+  return matchAndRewriteBinaryOp(op, rewriter, GetMinMaxRescaleScales);
+}
+
+LogicalResult matchAndRewriteOp(stablehlo::MaxOp op,
+                                PatternRewriter &rewriter) {
+  return matchAndRewriteBinaryOp(op, rewriter, GetMinMaxRescaleScales);
+}
+
+LogicalResult matchAndRewriteCompareOp(stablehlo::CompareOp op,
+                                       PatternRewriter &rewriter) {
+  Value lhs = op.getLhs();
+  Value rhs = op.getRhs();
+  Value result = op.getResult();
+
+  auto lhsType = cast<ShapedType>(lhs.getType());
+  auto rhsType = cast<ShapedType>(rhs.getType());
+  auto resultType = cast<ShapedType>(result.getType());
+
+  auto lhsQType =
+      dyn_cast<quant::UniformQuantizedType>(lhsType.getElementType());
+  auto rhsQType =
+      dyn_cast<quant::UniformQuantizedType>(rhsType.getElementType());
+
+  if (!lhsQType || !rhsQType) {
+    return rewriter.notifyMatchFailure(
+        op,
+        "The conversion supports operands with per-tensor quantized "
+        "types only");
+  }
+
+  double lhsScale = lhsQType.getScale();
+  double rhsScale = rhsQType.getScale();
+  double maxScale = std::max(lhsScale, rhsScale);
```

Comment:
Just curious to know why the max scale for add/sub is twice the max scale used for compare/max,min. Is that depends on some tosa backend details?

Reply:
I dont know for sure. my guess is add/sub of 2 numbers may double the values whereas min/max of two numbers would not?